include/dlaf/matrix/extra_buffers.h

```diff
@@ -0,0 +1,79 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <vector>
+
+#include "dlaf/blas/tile_extensions.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/types.h"
+
+namespace dlaf {
+
+template <class T, Device D>
+struct ExtraBuffers : protected Matrix<T, D> {
+  ExtraBuffers(const TileElementSize bs, const SizeType size)
+      : Matrix<T, D>{{bs.rows() * size, bs.cols()}, bs}, nbuffers_(size) {
+    namespace ex = pika::execution::experimental;
+    for (const auto& i : common::iterate_range2d(Matrix<T, D>::distribution().localNrTiles()))
+      ex::start_detached(Matrix<T, D>::readwrite_sender(i) |
+                         tile::set0(dlaf::internal::Policy<dlaf::DefaultBackend_v<D>>(
+                             pika::execution::thread_priority::high)));
+  }
+
+  auto read_sender(SizeType index) {
+    return Matrix<T, D>::read_sender(internalIndex(index));
+  }
+
+  auto readwrite_sender(SizeType index) {
+    return Matrix<T, D>::readwrite_sender(internalIndex(index));
+  }
+
+  template <class TileSender>
+  [[nodiscard]] auto reduce(TileSender tile) {
+    namespace ex = pika::execution::experimental;
+
+    std::vector<ex::any_sender<pika::shared_future<matrix::Tile<const T, D>>>> buffers;
```

Comment:
The type is so cumbersome, probably @msimberg is aware of this and is going to say that it is something we will manage when we will make the tile sender mechanism pika::future free.

I don't know if there is any workaround available, but AFAIK currently our `Unwrapping` facilities does not unwrap `vector`, so we just get it sent as it is even using our `dlaf::internal::transform`.

Reply:
I see your complaint, but `typedef` exists :wink: I see that complex eventually being a helper typedef inside e.g. `Matrix`, or at least in the `dlaf::matrix` namespace and then you'll be able to write `std::vector<ReadOnlyTileSender>` or something like that. Would that partially or completely take care of your concerns here? This is what I'm planning on doing in #766 (because currently it's also dealing with long unwieldy types).